_posts/2020-02-7-dns-based-discovery.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+name:  "DNS based discovery"
+title:  "DNS based discovery"
+date:   2020-02-7 12:00:00 +0100
+author: decanus
+published: true
+permalink: /dns-based-discovery
+categories: research
+summary: A look at EIP-1459 and the benefits of DNS based discovery.
+image: /assets/img/vac.jpg
+---
+
+Discovery in p2p networks is the process of how nodes find each other and specific resources they are looking for. Popular discovery protocols, such as [Kademlia](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf) which utilizes a [distributed hash table](https://en.wikipedia.org/wiki/Distributed_hash_table) or DHT, are highly inefficient for resource restricted devices, as we cannot expect a mobile phone for example to synchronize an entire DHT using cellular data.
+
+Another issue is how we do the initial bootstrapping. In other words, how does a client find its first node to then discover the rest of the network? In most applications, including Status right now, this is done with a [static list of nodes](https://github.com/status-im/specs/blob/master/status-client-spec.md#bootstrapping) that a client can connect to.
+
+So in summary, we have a static list that provides us with nodes we can connect to which then allows us to synchronize the rest of the network using something like Kademlia. But what we need is something that can easily be mutated, guarantees a certain amount of security, and is efficient for resource restricted devices. Ideally our solution would also be robust and scalable.
```

Comment:
>  So in summary, we have a static list that provides us with nodes we can connect to which then allows us to synchronize the rest of the network using something like Kademlia. But what we need is something that can easily be mutated, guarantees a certain amount of security, and is efficient for resource restricted devices. Ideally our solution would also be robust and scalable. 

First two sentences can probably be more direct. E.g. don't start with "So" and "But". Also "sync rest of network" is a bit too prescriptive, maybe something more like dynamically discover the rest of the network would work?